### PR TITLE
Stofs 2D Support

### PR DESCRIPTION
**Pull request by mpiannucci**
@omkar-334 Please test out this PR. 

When I download the data locally, the subset takes 4.0s, but when using the s3 version it takes a minute or minute and a half. 

I am happy to chat through these changes with you so you understand. The issue here is that the dimension ordering for the face node connectivity was reversed from the ordering that FVCOM uses, so we have to make the ugrid subsetter aware of this. 

I looked at the comparison with thalassa and I believe the difference in speed has to do with the reindexing and the the element selection but i am not sure. But this should give results that are more comparable I believe. 

#16 
#9

### CODE REVIEW

**Comment by omkar-334** [comment]
> I looked at the comparison with thalassa and I believe the difference in speed has to do with the reindexing and the the element selection but i am not sure. But this should give results that are more comparable I believe.
Got it, I have tested this out and it does indeed give more comparable results.  
Moreover, I noticed a considerable decrease in computation time when using fsspec rather than s3fs.

File: xarray_subset_grid/grids/ugrid.py [hunk 1]
```diff
@@ -113,31 +112,54 @@ def subset_polygon(
         x_var, y_var = mesh.node_coordinates.split(" ")
         x, y = ds[x_var], ds[y_var]
 
+        # NOTE: When the first dimension is "nele", the face_node_connectivity
```
Comment (ChrisBarker-NOAA) [inline xarray_subset_grid/grids/ugrid.py:115]: According to the UGRID spec:
"""
The connectivity array will thus be a matrix of size nFaces x 3. 
"""
so the "nele" dimension should always be first in a compliant file.

If we are finding files that. need to be transposed (ADCIRC?) -- what to do?

I think that should be handled by a pre-processor -- the make compliant function:

`assign_ugrid_topology`

If we are consistent about that, then we'll save a lot if blocks in the core code, like here.

Comment (mpiannucci) [reply xarray_subset_grid/grids/ugrid.py:115]: I do not think the preprocessor should mess with the dataset itself personally, but I can be convinced maybe.

Yes the ADCIRC files have reversed dimensions on the face node connectivity, which could happen with any dataset really.

Comment (ChrisBarker-NOAA) [reply xarray_subset_grid/grids/ugrid.py:115]: You are probably right -- as much as I want to be pedantic about the standard, we shouldn't actually change anything (as opposed to adding metadata) in the process unless the user asks for it ...

**Pull request merged**